### PR TITLE
Group 2 Solution

### PR DESCRIPTION
**Pull request by jivewade**
Vince Kumar
Justin Wade
Steven Martinez

### CODE REVIEW

**Comment by martinzugnoni** [comment]
All tests passing, congrats! 🎉

File: calculator/main.py [hunk 1]
```diff
@@ -13,21 +13,44 @@ def create_new_calculator(operations=None):
...
     """
-    pass
-
+    
+    op = calc['operations'][operation]
+    # Error is somewhere in the if loops
+    if operation not in calc['operations'].keys():
+        raise InvalidOperation("%s is invalid." % operation)
+    
+    try:
```
Comment (koakekuna) [inline calculator/main.py:41]: Good use of try block. Wish I had thought of that.

File: calculator/main.py [hunk 2]
```diff
@@ -58,18 +84,21 @@ def get_history(calc):
...
 
 def repeat_last_operation(calc):
     """
     Returns the result of the last operation executed in the history.
     """
-    pass
+    history = calc['history']
+    if not history:
+        return None
+    return history[-1][-1]
```
Comment (koakekuna) [inline calculator/main.py:104]: This one bugged me for awhile. I eventually came up with history[-1][3] but I like yours better.

File: calculator/operations.py [hunk 3]
```diff
@@ -1,19 +1,24 @@
...
 def add(*args):
-    # your implementation here
-    pass
-
+    return sum(args)
+    
 def subtract(*args):
-    # your implementation here
-    pass
+    return args[0] - sum(args[1:])
```
Comment (koakekuna) [inline calculator/operations.py:6]: Wow, awesome way to write this subtract function. We difficulty figuring out how to deal with the *args.

File: calculator/main.py [hunk 4]
```diff
@@ -13,21 +13,44 @@ def create_new_calculator(operations=None):
...
+    op = calc['operations'][operation]
+    # Error is somewhere in the if loops
+    if operation not in calc['operations'].keys():
+        raise InvalidOperation("%s is invalid." % operation)
+    
+    try:
+        assert isinstance(params, tuple)
+        assert all([isinstance(x, (int, float)) for x in params])
+    except AssertionError:
+        raise InvalidParams("Given params are invalid.")
```
Comment (aroni11) [inline calculator/main.py:45]: Very nice way to check for errors.

**Pull request closed without merging**